src/freenet/node/updater/NodeUpdater.java

```diff
@@ -91,15 +91,23 @@ public abstract class NodeUpdater implements ClientGetCallback, USKCallback, Req
 	}
 
 	void start() {
+        try {
+            this.subscribe();
+		} catch(MalformedURLException e) {
+            manager.blow("The auto-update URI isn't valid and can't be used", true);
+        }
+	}
+
+    void subscribe() throws MalformedURLException {
 		try {
 			// because of UoM, this version is actually worth having as well
 			USK myUsk = USK.create(URI.setSuggestedEdition(currentVersion));
 			core.getUskManager().subscribe(myUsk, this, true, getRequestClient());
 		} catch(MalformedURLException e) {
 			Logger.error(this, "The auto-update URI isn't valid and can't be used");
-			manager.blow("The auto-update URI isn't valid and can't be used", true);
+            throw e;
```

Comment:
This irks me… catching an exception, just to log it, then re-throw it and ignore it in one of the methods. That feels like a lot of work.

Hmm, oh boy, here I go, making it overly complicated again… but what about:

```
private void subscribe(Runnable onError) {
    try {
    } catch (MalformedURLException e) {
        Logger.error(this, "The auto-update URI isn't valid and can't be used");
        onError.run();
    }
}
```
so we can have
```
void start() {
    subscribe(() -> manager.blow("The auto-update URI isn't valid and can't be used", true));
}
```
and
```
public void onChangeURI(FreenetURI uri) {
    kill();
    this.URI = uri;
    subscribe(() -> {});
    maybeUpdate();
}
```
?

Reply:
I wanted to say “no, this makes it too non-obvious”, but it’s actually much nicer with the Runnable.

So: yes ☺

Thank you!